### PR TITLE
Return html for a new conditional

### PR DESCRIPTION
**Pull request by tomas-stefano**
When the client-side JS requests a new conditional we should
return sending the right name parameter.

So let's assume that on the adding branch page there are 3 conditionals
and the user clicks "Add another rule" / "Add another conditional" (I am
not sure what is the name of the button) then the JS will request
a new conditional sending the total amount of conditionals that exists
withing the page

```
/services/123/branches/456/conditionals/3 (3 conditionals)
```

and this will return an HTML that needs to be rendered within the page.

### CODE REVIEW

File: app/views/branches/_form_conditionals.html.erb [hunk 1]
```diff
@@ -1,4 +1,4 @@
-<%= f.fields_for :conditionals do |conditional| %>
+<%= f.fields_for :conditionals, child_index: conditional_index do |conditional| %>
   <div class="govuk-form-group <%= conditional.object.errors[:next].empty? ? '' : 'govuk-form-group--error'  %>" data-index-conditional="<%= conditional.index %>">
```
Comment (brenetic) [inline app/views/branches/_form_conditionals.html.erb:2]: would now be a good time to swap `data-index-conditional` for `data-conditional-index`?

same for the expressions index

File: config/routes.rb [hunk 2]
```diff
@@ -19,6 +19,8 @@
       resources :publish, only: [:index, :create]
       resources :pages, param: :page_uuid, only: [:create, :edit, :update, :destroy]
       resources :branches, param: :previous_flow_uuid, only: [:create, :edit] do
+        get '/conditionals/:conditional_index', to: 'branches#new_conditional'
```
Comment (brenetic) [inline config/routes.rb:22]: is it possible to put this in the `api` namespace?

File: app/controllers/branches_controller.rb [hunk 3]
```diff
@@ -22,10 +27,22 @@ def edit
...
+  def assign_branch
+    @branch = Branch.new(branch_attributes)
+    @branch.conditionals << Conditional.new(expressions: [OpenStruct.new])
+  end
+
   def branch_attributes
     {
       service: service,
-      previous_flow_uuid: params[:previous_flow_uuid] || params.require(:branch).permit(:previous_flow_uuid)[:previous_flow_uuid]
+      previous_flow_uuid: params[:previous_flow_uuid] || params[:branch_previous_flow_uuid] || params.require(:branch).permit(:previous_flow_uuid)[:previous_flow_uuid]
```
Comment (brenetic) [inline app/controllers/branches_controller.rb:45]: method?

**Comment by brenetic** [comment]
can we add some tests for this endpoint?

**Comment by tomas-stefano** [comment]
@brenetic all comments addressed! :)

**Review by brenetic — APPROVED** [review]
🚀

**Pull request merged**